config/config.go

```diff
@@ -175,28 +171,26 @@ func (c *Config) HttpServerConfig() httpServerConfig {
 	return httpConfig
 }
 
-func (c *Config) OracleConfig() oracle {
-	oracleConfig := c.Oracle
-
+func (c *Config) OracleAPIkeyReload(oracle Oracle) Oracle {
```

Comment:
This needs to updated so it is future proof.
It should do it by domain and source for example.
```
Eg.
0_ETHERSCAN_API_KEY
```

Reply:
hmm,, I remember you told me that each oracle service will only apprear once in that domain list?, why still need a domainID as prefix? in my understanding `ETHERSCAN_API_KEY` is the key from real etherscan, not for `etherscan` impl inside fee oracle

Reply:
But then we have to hardcode what exactly we are supporting which is the thing we want to avoid,